### ISSUE TITLE
Transpiling specific file panics

### ISSUE THREAD

**Issue by crlf0710** [immunant/c2rust#146]
The current master version panics when transpiling [this file](https://github.com/antirez/redis/blob/unstable/src/redis-check-rdb.c); it would be nice to fix it or output some diagnostics to help workaround it.
```
thread 'main' panicked at 'Could not find CTypeId(9443) in TypedAstContext', /mnt/m/Apps/.cargo/registry/src/mirrors.ustc.edu.cn-61ef6e0cd06fb9b8/c2rust-transpile-0.12.0/src/c_ast/mod.rs:766:21
stack backtrace:
   0: backtrace::backtrace::libunwind::trace
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/libunwind.rs:88
   1: backtrace::backtrace::trace_unsynchronized
             at /cargo/registry/src/github.com-1ecc6299db9ec823/backtrace-0.3.29/src/backtrace/mod.rs:66
   2: std::sys_common::backtrace::_print
             at src/libstd/sys_common/backtrace.rs:47
   3: std::sys_common::backtrace::print
             at src/libstd/sys_common/backtrace.rs:36
   4: std::panicking::default_hook::{{closure}}
             at src/libstd/panicking.rs:198
   5: std::panicking::default_hook
             at src/libstd/panicking.rs:212
   6: std::panicking::rust_panic_with_hook
             at src/libstd/panicking.rs:475
   7: std::panicking::continue_panic_fmt
             at src/libstd/panicking.rs:382
   8: std::panicking::begin_panic_fmt
             at src/libstd/panicking.rs:337
   9: c2rust_transpile::c_ast::TypedAstContext::get_src_loc
  10: <c2rust_transpile::translator::comments::CommentLocator as c2rust_transpile::c_ast::iterators::NodeVisitor>::pre
  11: c2rust_transpile::c_ast::iterators::NodeVisitor::visit_tree
  12: c2rust_transpile::translator::comments::<impl c2rust_transpile::translator::Translation>::locate_comments
  13: c2rust_transpile::translator::translate
  14: c2rust_transpile::transpile_single
  15: <core::iter::adapters::Map<I,F> as core::iter::traits::iterator::Iterator>::fold
  16: <alloc::vec::Vec<T> as alloc::vec::SpecExtend<T,I>>::from_iter
  17: c2rust_transpile::transpile
  18: c2rust_transpile::main
  19: std::rt::lang_start::{{closure}}
  20: std::rt::lang_start_internal::{{closure}}
             at src/libstd/rt.rs:49
  21: std::panicking::try::do_call
             at src/libstd/panicking.rs:294
  22: __rust_maybe_catch_panic
             at src/libpanic_unwind/lib.rs:82
  23: std::panicking::try
             at src/libstd/panicking.rs:273
  24: std::panic::catch_unwind
             at src/libstd/panic.rs:388
  25: std::rt::lang_start_internal
             at src/libstd/rt.rs:48
  26: main
  27: __libc_start_main
  28: _start
```

**Comment by rinon** [comment]
Thanks for the report. Yes, better diagnostics are definitely in order here. After investigating, it looks like this is due to lack of support for `__atomic_*` GCC builtins. They shouldn't be too hard to support, so I'll give it a try.

**Comment by rinon** [comment]
Ah looks like this is the same issue as #134

**Comment by rinon** [comment]
Should be fixed by 8d7d60be7931138814fde3b366452fee6755b31c, please reply if it still doesn't work for you.

**Comment by crlf0710** [comment]
Mmm, i updated c2rust code, and saw it's something different. It still doesn't work, but this time there's some diagnostics:
````
warning: Missing type 140736641804896 for node: AstNode { tag: TagFieldDecl, children: [], loc: SrcSpan { fileid: 35, begin_line: 1149, begin_column: 5, end_line: 1149, end_column: 23 }, type_id: Some(140736641804896), rvalue: LValue, macro_expansions: [], extras: [String("stat_net_output_bytes"), Null, U64(10112), U64(64)] }
Exported Clang AST was invalid. Check warnings above for unimplemented features.
--> /mnt/l/ThirdParty/redis/src/server.h:1149:5
 [-Wclang-ast]
````
